server/api_server/schemas.py

```
@@ -0,0 +1,670 @@
+    createNew: bool = Field(False, description="Create new device if not exists")
+
+
+class DeleteDevicesRequest(BaseModel):
+    """Request to delete multiple devices."""
+    macs: List[str] = Field([], description="List of MACs to delete")
+    confirm_delete_all: bool = Field(False, description="Explicit flag to delete ALL devices when macs is empty")
+
+    @model_validator(mode="after")
+    def check_delete_all_safety(self) -> DeleteDevicesRequest:
+        if not self.macs and not self.confirm_delete_all:
+            raise ValueError("Must provide at least one MAC or set confirm_delete_all=True")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)